### PR TITLE
Harden disk_free_space check in CheckSetupController

### PR DESCRIPTION
**Pull request by solracsf**
To avoid things like https://github.com/nextcloud/server/issues/26034

### CODE REVIEW

**Review by MichaIng — APPROVED** [review]

File: apps/settings/lib/Controller/CheckSetupController.php [hunk 1]
```diff
@@ -769,14 +769,14 @@ protected function isEnoughTempSpaceAvailableIfS3PrimaryStorageIsUsed(): bool {
...
 		}
 
-		$tempPath = sys_get_temp_dir();
+		$tempPath = ini_get('upload_tmp_dir') ? ini_get('upload_tmp_dir') : sys_get_temp_dir();
 		if (!is_dir($tempPath)) {
 			$this->logger->error('Error while checking the temporary PHP path - it was not properly set to a directory. value: ' . $tempPath);
 			return false;
 		}
-		$freeSpaceInTemp = disk_free_space($tempPath);
+		$freeSpaceInTemp = function_exists('disk_free_space') ? disk_free_space($tempPath) : false;
```
Comment (CarlSchwan) [inline apps/settings/lib/Controller/CheckSetupController.php:777]: Can we return another error if the `disk_free_space` function is not available? Telling the user that there is no free disk space is not completely correct. Also maybe we shouldn't hard fail on this, but not completely sure

File: apps/settings/lib/Controller/CheckSetupController.php [hunk 2]
```diff
@@ -769,14 +769,14 @@ protected function isEnoughTempSpaceAvailableIfS3PrimaryStorageIsUsed(): bool {
 			return true;
 		}
 
-		$tempPath = sys_get_temp_dir();
+		$tempPath = ini_get('upload_tmp_dir') ? ini_get('upload_tmp_dir') : sys_get_temp_dir();
```
Comment (artonge) [inline apps/settings/lib/Controller/CheckSetupController.php:772]: Not sure about this one. Will `$tempPath` be used to upload things? If no we should keep the old line.

File: apps/settings/lib/Controller/CheckSetupController.php (hunk 1 above)
Comment (artonge) [reply apps/settings/lib/Controller/CheckSetupController.php:777]: Error message is below

File: apps/settings/lib/Controller/CheckSetupController.php (hunk 2 above)
Comment (acsfer) [reply apps/settings/lib/Controller/CheckSetupController.php]: Here is my point; in our (shared) systems, /tmp (the 99% returned folder of `sys_get_temp_dir`) is not world RW, while each user gets a private `upload_tmp_dir` set by `php_admin_flag`. But as this one could also be set like `php_admin_value sys_temp_dir` let's keep it the old way.

File: apps/settings/lib/Controller/CheckSetupController.php (hunk 1 above)
Comment (acsfer) [reply apps/settings/lib/Controller/CheckSetupController.php:777]: Error message improved (grammar)

**Comment by szaimen** [comment]
@solracsf please fix dco and can you squash this? Thanks!

**Comment by szaimen** [comment]
> @solracsf please fix dco and can you squash this? Thanks!

Otherwise we cannot merge this...

**Review by MichaIng — APPROVED** [review]

**Comment by artonge** [comment]
Any reason this was closed @solracsf ?

**Comment by solracsf** [comment]
8 months and a simple squash blocking a PR...feel free to reopen, squash and merge.

File: apps/settings/lib/Controller/CheckSetupController.php [hunk 3]
```diff
@@ -392,7 +392,7 @@ private function isCorrectMemcachedPHPModuleInstalled() {
 	 */
 	private function isSettimelimitAvailable() {
 		if (function_exists('set_time_limit')
-			&& strpos(@ini_get('disable_functions'), 'set_time_limit') === false) {
+			&& strpos(ini_get('disable_functions'), 'set_time_limit') === false) {
```
Comment (szaimen) [inline apps/settings/lib/Controller/CheckSetupController.php:395]: what if ini_get is disabled?

Comment (artonge) [reply apps/settings/lib/Controller/CheckSetupController.php:395]: Is this possible ?

**Review by artonge — APPROVED** [review]

**Pull request merged**